### PR TITLE
Generate functions upcasts

### LINKED ISSUE #36: Trait constaints
Which better set of names for types in constraint: T, U, V, W, X,Y,Z or T, T2, T3,... ?

### PR DESCRIPTION
**Pull request by EPashkin**
Closes #36

### CODE REVIEW

File: src/analysis/upcasts.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
+use std::collections::vec_deque::VecDeque;
+use std::slice::Iter;
+use std::vec::Vec;
+
+pub struct Upcasts {
+    unused: VecDeque<String>,
+    used: Vec<(String, String, String)>,
```
Comment (gkoz) [inline src/analysis/upcasts.rs]: I'd appreciate a comment describing the meanings of the tuple elements.

**Comment by EPashkin** [comment]
`//Vector tuples <parameter name>, <alias>, <type>` ?

**Comment by gkoz** [comment]
> `//Vector tuples <parameter name>, <alias>, <type>` ?

Yes, that'd be fine.

**Comment by EPashkin** [comment]
Done

**Comment by gkoz** [comment]
Thanks.

**Pull request merged**